app/main.py

```
@@ -20,8 +22,37 @@ def sha256_hash_str(to_hash: str) -> str:
     return sha256(to_hash.encode("utf-8")).hexdigest()
 
 
+def brute_force_range(start: int, end: int, target_hashes: set[str]) -> dict[str, str]:
+    found = {}
+    for num in range(start, end):
+        candidate = str(num).zfill(8)
+        hash_val = sha256_hash_str(candidate)
+        if hash_val in target_hashes:
+            found[hash_val] = candidate
+            if len(found) == len(target_hashes):
+                break
+    return found
+
+
 def brute_force_password() -> None:
-    pass
+    cpu_count = max(1, multiprocessing.cpu_count())
+    chunk_size = 100000000 // cpu_count
+    target_hashes = set(PASSWORDS_TO_BRUTE_FORCE)
+
+    with ProcessPoolExecutor(cpu_count) as executor:
+        futures = []
+        for i in range(cpu_count):
+            start = i * chunk_size
+            end = (i + 1) * chunk_size if i < cpu_count - 1 else 100_000_000
+            futures.append(executor.submit(brute_force_range, start, end, target_hashes))
+
+        results = {}
+        for future in futures:
+            results.update(future.result())
+
+    for h in PASSWORDS_TO_BRUTE_FORCE:
+        print(results[h])
```

Risk: `print(results[h])` assumes every target hash exists in `results`. If any hash is missing (for example due to a bug or a worker failure), this will raise KeyError. Validate presence (e.g., `if h not in results: raise RuntimeError(...)`) or fail fast with a clear message. Also ensure you print exactly the required 10 plaintexts (one per line) before any optional output.